py/packages/corpora_cli/commands/workon.py

```diff
@@ -0,0 +1,98 @@
+from corpora_client.models.message_schema import MessageSchema
+from corpora_cli.context import ContextObject
+
+app = typer.Typer(help="Interactive issue creation CLI")
+
+session = PromptSession()
+
+
+@app.command()
+def file(ctx: typer.Context, path: str):
+    """
+    Workon a file in the corpus.
+    """
+    c: ContextObject = ctx.obj
+    c.console.print(f"Working on file: {path}", style="bold blue")
+    ext = path.split(".")[-1]
+
+    # show current file content on disk, load the content from the
+    # CWD and print it with dim
+    with open(path, "r") as f:
+        current_file_content = f.read() if f else ""
+    c.console.print("Current file content:", style="bold green")
+    c.console.print(current_file_content, style="dim")
+
+    # Start with an empty message list
+    messages: List[MessageSchema] = []
+
+    if current_file_content:
+        messages.append(
+            MessageSchema(
+                role="user",
+                text=f"The original file content was:\n```{ext}\n{current_file_content}```",
+            )
+        )
+
+    # REPL loop
+    while True:
+        user_input = session.prompt(
+            ("What to do?\n" if not messages else "How to revise?\n"),
+            multiline=True,
+            vi_mode=True,
+        )
+
+        if not user_input:
+            c.console.print("No input provided. Please try again.", style="yellow")
+            continue
+
+        # Add the user's input as a new message
+        messages.append(MessageSchema(role="user", text=user_input.strip()))
+
+        # Send the current messages to generate a draft issue
+        c.console.print("Generating revision...", style="bold blue")
+
+        # if file doesn't exist, use empty string
+        with open(".corpora/VOICE.md", "r") as f:
+            voice = f.read() if f else ""
+        with open(".corpora/PURPOSE.md", "r") as f:
+            purpose = f.read() if f else ""
+        with open(".corpora/STRUCTURE.md", "r") as f:
+            structure = f.read() if f else ""
+        with open(f".corpora/{ext}/DIRECTIONS.md", "r") as f:
+            directions = f.read() if f else ""
+
+        # c.console.print(voice, style="dim")
+        # c.console.print(purpose, style="dim")
+        # c.console.print(structure, style="dim")
+        # c.console.print(directions, style="dim")
+
+        revision = c.workon_api.file(
+            CorpusFileChatSchema(
+                messages=messages,
+                corpus_id=c.config["id"],
+                path=path,
+                voice=voice,
+                purpose=purpose,
+                structure=structure,
+                directions=directions,
+            )
+        )
+        c.console.print(f"{revision}", style="dim")
+        c.console.print(f"{path}", style="dim magenta")
+
+        if typer.confirm("Write file?"):
+            with open(path, "w") as f:
+                f.write(revision)
+            c.console.print("File written!", style="green")
```

Comment:
Here it might be cool to keep the current messages context and change to a different file? Or give the user an opportunity to continue with the context that has been built.